nni/compression/pytorch/quantization_speedup/integrated_tensorrt.py

```diff
@@ -0,0 +1,296 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import time
+import tensorrt as trt
+import numpy as np
+import torch
+
+from . import frontend_to_onnx as fonnx
+from . import calibrator as calibrator
+from . import trt_pycuda as common
+from .backend import BaseModelSpeedup
+
+# TRT_LOGGER = trt.Logger(trt.Logger.VERBOSE)
+TRT_LOGGER = trt.Logger()
+
+class CalibrateType:
+    LEGACY = trt.CalibrationAlgoType.LEGACY_CALIBRATION
+    ENTROPY = trt.CalibrationAlgoType.ENTROPY_CALIBRATION
+    ENTROPY2 = trt.CalibrationAlgoType.ENTROPY_CALIBRATION_2
+    MINMAX = trt.CalibrationAlgoType.MINMAX_CALIBRATION
+
+Precision_Dict = {
+    8: trt.float32,
+    16: trt.float16,
+    32: trt.float32
+}
+
+def build_engine(model_file, calib, config=None, extra_layer_bit=32, strict_datatype=False):
+    with trt.Builder(TRT_LOGGER) as builder, builder.create_network(common.EXPLICIT_BATCH) as network, \
+        trt.OnnxParser(network, TRT_LOGGER) as parser:
+        """
+        This function builds an engine from an onnx model with calibration process.
+        """
+        # Attention that, builder should be set to 1 because of the implementation of allocate_buffer
+        builder.max_batch_size = 1
+        builder.max_workspace_size = common.GiB(1)
+
+        if extra_layer_bit == 32 and config is None:
+            pass
+        elif extra_layer_bit == 8 and config is None:
```

Comment:
what is `extra_layer_bit` is 16 and config is None?

Reply:
We should turn on fp16 mode. Have modified.